src/components/course-calendar/plan.cjsx

```diff
@@ -120,15 +130,22 @@ CoursePlan = React.createClass
     @setState(publishState)
 
   componentWillMount: ->
-    @subscribeToPublishing(@props.item.plan)
-    @syncRoute()
+    {plan} = @props.item
+    publishState = PlanHelper.subscribeToPublishing(plan, @checkPublishingStatus)
+    isViewingStats = @_doesPlanMatchesRoute()
+
+    state = _.extend({isViewingStats}, publishState)
+    @setState(state)
+
     location = @context.router.getLocation()
     location.addChangeListener(@syncRoute)
 
   componentWillReceiveProps: (nextProps) ->
     if @props.item.plan.id isnt nextProps.item.plan.id
-      @subscribeToPublishing(nextProps.item.plan)
+      publishState = PlanHelper.subscribeToPublishing(nextProps.item.plan, @checkPublishingStatus)
       @stopCheckingPlan(@props.item.plan)
+      state = _.extend(@getStateByProps(nextProps), publishState)
```

Comment:
If a prop other than plan was updated, `nextProps.plan` will be the same as `@props.plan`.

In that case, this will subscribe and then immediately unsubscribe from it when `@stopCheckingPlan` is called.  Maybe `stopCheckingPlan` should be called first?